tests/api/printf.cpp

```diff
@@ -104,64 +115,131 @@ static std::string getStdoutFileName(temp_folder_deletion& temp) {
+                            &platform_name_len);
+    RETURN_ON_FAILURE(err, nullptr);
 
-    size_t gws = 1;
-    size_t lws = 1;
-    EnqueueNDRangeKernel(kernel, 1, nullptr, &gws, &lws, 0, nullptr, nullptr);
-    Finish();
+    std::string platform_name(platform_name_len - 1, ' ');
+    err = clGetPlatformInfo(platform, CL_PLATFORM_NAME, platform_name_len,
+                            const_cast<char*>(platform_name.data()), nullptr);
+    RETURN_ON_FAILURE(err, nullptr);
 
-    releaseStdout(fd);
-    auto printf_buffer = getStdoutContent();
-    ASSERT_NE(printf_buffer, nullptr);
+    std::cout << "Platform: " << platform_name << std::endl;
 
-    ASSERT_STREQ(printf_buffer, message);
+    return platform;
 }
 
-TEST_F(WithCommandQueue, TooLongPrintf) {
-    // each print takes 12 bytes (4 for the printf_id, and 2*4 for the 2 integer
-    // to print) + 4 for the byte written counter
-    auto cfg1 =
-        CLVK_CONFIG_SCOPED_OVERRIDE(printf_buffer_size, uint32_t, 28, true);
+cl_device_id get_device_test(cl_platform_id platform) {
+    cl_device_id device;
+    cl_int err;
 
-    temp_folder_deletion temp;
-    stdoutFileName = getStdoutFileName(temp);
+    err = clGetDeviceIDs(platform, CL_DEVICE_TYPE_ALL, 10, &device, nullptr);
+    RETURN_ON_FAILURE(err, nullptr);
 
-    int fd;
-    ASSERT_TRUE(getStdout(fd));
+    size_t device_name_len;
+    err = clGetDeviceInfo(device, CL_DEVICE_NAME, 0, nullptr, &device_name_len);
+    RETURN_ON_FAILURE(err, nullptr);
+
+    std::string device_name(device_name_len - 1, ' ');
+    err = clGetDeviceInfo(device, CL_DEVICE_NAME, device_name_len,
+                          const_cast<char*>(device_name.data()), nullptr);
+    RETURN_ON_FAILURE(err, nullptr);
+
+    std::cout << "Device: " << device_name << std::endl;
+
+    return device;
+}
+
+std::string captured_output; // Global to store the captured output
+
+/* Define a printf callback function. */
+void printf_callback(const char* buffer, size_t len, size_t complete,
+                     void* user_data) {
+    captured_output.append(buffer, len);
+}
+
+void get_command_queue_printf(cl_command_queue& queue, cl_context new_ctx,
+                              cl_device_id device) {
+    cl_int err;
+    queue = clCreateCommandQueue(new_ctx, device, 0, &err);
+    ASSERT_CL_SUCCESS(err);
+}
+
+void run_kernel_print_context(const char* source, const char* name) {
+    cl_int err;
+    auto platform = get_platform_test();
+    auto device = get_device_test(platform);
+    cl_context_properties properties[] = {
+        /* Enable a printf callback function for this context. */
+        CL_PRINTF_CALLBACK_ARM, (cl_context_properties)printf_callback,
+
+        /* Request a minimum printf buffer size of 4MiB for devices in the
+        context that support this extension. */
+        CL_PRINTF_BUFFERSIZE_ARM, (cl_context_properties)0x100000,
+
+        CL_CONTEXT_PLATFORM, (cl_context_properties)gPlatform, 0};
```

Comment:
I understand that this is part of the sample example [here](https://registry.khronos.org/OpenCL/extensions/arm/cl_arm_printf.html), but I don't think it is needed.